crates/rolldown/src/ecmascript/format/utils/namespace.rs

```diff
@@ -28,29 +28,20 @@ use rolldown_utils::ecma_script::is_validate_assignee_identifier_name;
 ///    ```
 fn generate_namespace_definition(name: &str) -> (String, String) {
   let parts: Vec<&str> = name.split('.').collect();
+  let mut initial_code = String::new();
+  let mut final_code = String::from("this");
 
-  let initialization_code = parts
-    .iter()
-    .enumerate()
-    .scan(String::new(), |state, (i, part)| {
-      // We use `scan` to generate the declaration sentence level-by-level.
-      let caller = generate_caller(part);
-      state.push_str(&caller);
-      let line = if i < parts.len() - 1 {
-        Some(format!("this{state} = this{state} || {{}};\n"))
-      } else {
-        None
-      };
-      Some(line)
-    })
-    .flatten()
-    .collect::<String>();
+  for (i, part) in parts.iter().enumerate() {
+    let caller = generate_caller(part);
+    final_code.push_str(&caller);
 
-  // TODO do not call the `generate_caller` function twice.
-  let final_code =
-    format!("this{}", parts.iter().map(|&part| generate_caller(part)).collect::<String>());
+    if i < parts.len() - 1 {
+      let callers = &final_code[4..];
```

Comment:
Could you please clarify if this is related to the `this` keyword? It would be preferable to avoid hardcoding this, as we need to support different namespace roots.

Reply:
Yes, because I pre-set `this` in `final_code`. Do I need to do anything else?

Reply:
It is advisable to modify the code to check the length of the `this` variable instead of hardcoding. We may support the UMD format, which uses `global` instead of `this`. Also [here](https://rollupjs.org/configuration-options/#context).